### ISSUE TITLE
Text-to-speech stops working sometimes and restart is needed

### ISSUE THREAD

**Issue by madnik7** [ankidroid/Anki-Android#4043]
I switch between applications many times, but there are many times that AnkiDroid suddenly stop reading text and it wrote: **"Initialing TTS"** after around 2 second it wrote **"No text-to-speech available (eng)"**

So I need to
1) I need to close the AnkiDroid and start it again.
2) Sometimes the first method does not solve the problem and I need to close ankidroid and go to "android settings > Language input >  text-to-speech options" then it solves the problem.

I think that might be difficult to reproduce the problem, but I found this problem always happens if you open a card and in the middle of reading (TTS) you just press the "night mode". I hope this could be a key to find the first problem too.

My Android Device is: Samsung Galaxy Note 4 (SM-N910C) (Android 5.1.1)
I have tested with Google TTS and IVONA TTS engine, the problem exists on both TTS engine,

**Comment by xyzzklk** [comment]
I'm having the exact same problem!
https://user-images.githubusercontent.com/90095370/223718916-7074b079-a0b9-44bb-8b35-ce4ebc582a1a.mp4

![IMG_20230308_205152.jpg](https://user-images.githubusercontent.com/90095370/223719370-d39351df-d357-4ba6-a3e6-37d9c7fbf6cf.jpg)

**Comment by david-allison** [comment]
Thanks for the report. Could you provide Debug Info: https://ankidroid.org/help.html#_ankidroid_2_16

**Comment by xyzzklk** [comment]
Sure!
```
AnkiDroid Version = 2.15.6

Android Version = 8.0.0

Manufacturer = HUAWEI

Model = HUAWEI NXT-TL00

Hardware = hi3650

Webview User Agent = Mozilla/5.0 (Linux; Android 8.0.0; HUAWEI NXT-TL00 Build/HUAWEINXT-TL00; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/110.0.5481.153 Mobile Safari/537.36

ACRA UUID = 894f29ca-0fa3-412b-b621-9888593ce480

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true
```

**Comment by david-allison** [comment]
Debug info won't change. Thanks!

**Comment by xyzzklk** [comment]
Got it! You are welcome.

**Comment by xyzzklk** [comment]
The bug occurred again in the version `2.16alpha94`.

However, it occurred immediately right after I did some operations. I don't know whether it was because of my operations or it is just a randomly shown behavior.

I am using a note type derived from the `Basic (and reversed card)` note type. I customized the templates. The template of the `Back` of `Card 1` was customized by my self, it was something like
```
<tts service="android" voice="en_US">
<b>{{Front}}</b><br>
{{Back}}
</tts>
```
And after I changed it to
```
<tts service="android" voice="en_US">
<b>{{Front}}</b>
</tts>

<br>

<tts service="android" voice="en_US">{{Back}}</tts>
```
because I do not want the system to read `{{Front}}` and `{{Back}}` continuously as one sentence, the bug occurred.

For the purpose of testing the behaviors without restarting the app, I reset the languages and did a force full sync and then that behavior disappeared (maybe just temporarily, I do not know). I did that because I thought that changing card templates without force full syncing might be dangerous.

Finally, here is the new debug info that might help.

```
AnkiDroid Version = 2.16alpha94

Android Version = 8.0.0

Manufacturer = HUAWEI

Model = HUAWEI NXT-TL00

Hardware = hi3650

Webview User Agent = Mozilla/5.0 (Linux; Android 8.0.0; HUAWEI NXT-TL00 Build/HUAWEINXT-TL00; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/110.0.5481.153 Mobile Safari/537.36

ACRA UUID = 894f29ca-0fa3-412b-b621-9888593ce480

New schema = false

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true
```

**Comment by xyzzklk** [comment]
There is something new that I have just found, but I don't know if it means something.

**Important Information**
I mentioned my templates before, so I am not going to describe them again. However, to make things clear, I got `<tts (attributes...)></tts>` tags on the back of both `Card1` and `Card2`. Back is the answer side. So every time I flip a card (to display the answer), Anki should read something out using Text-To-Speech. And I am using the automatically play audio settings in deck options.

**The Pattern**
Every time, if I display the answer (flip a card) then lock my screen or if I switch to some other things (switching the interface to another app/task, a different interface in Anki such as that example that I wanted to edit card templates during studying; or something like leaving Anki in the background) for around 10-15 seconds and then come back, the TTS will stop working. Then if I try to study the next card and flip it, it will fail and show this error because the TTS has already stopped working.

[Here](https://user-images.githubusercontent.com/90095370/224320408-0b704080-a972-4047-934d-bdbe277363d8.mp4) is a video, showing that the TTS engine fails after hanging Anki in the background. The failure appears when trying to reopen it.

**Conclusion**
Long story short, at least for now the conclusion drawn from my test is, _TTS engine stops working after switching the interface or leaving AnkiDroid in the background_. To make it start working again, you will need to restart the app or quit the studying interface and then reenter it.

The behavior is almost exactly the same as @madnik7 's description, although TTS might not crash if you just leave Anki for about 1-3 seconds and if it did crashed I do not need to modify TTS in the settings of Android to make it work again.
